src/Disks/DiskBackup.h

```diff
@@ -0,0 +1,206 @@
+#pragma once
+
+#include <Disks/IDisk.h>
+#include <Disks/ObjectStorages/IMetadataStorage.h>
+#include <IO/ReadBufferFromFileBase.h>
+#include <IO/WriteBufferFromFile.h>
+#include "Common/Exception.h"
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int NOT_IMPLEMENTED;
+}
+
+class DiskBackup : public IDisk
```

Comment:
It will be good to also add a comment before this class definition with a small explanation of the purpose of this class. A sentence about its purpose from https://github.com/ClickHouse/ClickHouse/issues/42154 will suffice.